### PR TITLE
Add autoupdate

### PR DESCRIPTION
**Pull request by gferraro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ These recordings will be analysed to detect the volume of birdsong and how it ch
 This project is licensed under the Affero General Public License
 (https://www.gnu.org/licenses/agpl-3.0.en.html).
 
+## Autoupdates
+
+* In order to allow bird monitor to auto update, the APK must be installed from the github releases page (https://github.com/TheCacophonyProject/bird-monitor/releases)
```
Comment (mjs) [inline README.md]:
```
In order to allow bird monitor to auto update:
* the APK must be installed from the github releases page (https://github.com/TheCacophonyProject/bird-monitor/releases)
* ...
```

File: app/src/main/java/nz/org/cacophony/birdmonitor/Util.java [hunk 2]
```diff
@@ -815,7 +854,7 @@ public static void createFailSafeAlarm(Context context) { // Each alarm creates
 
     public static long getMilliSecondsBetweenRecordings(Prefs prefs) {
         if (prefs.getUseVeryFrequentRecordings()) {
-            return (long) prefs.getTimeBetweenVeryFrequentRecordingsSeconds();
+            return (long) prefs.getTimeBetweenVeryFrequentRecordingsSeconds() * 1000;
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Util.java:857]: Why this change? The method still says "seconds"

File: app/src/main/java/nz/org/cacophony/birdmonitor/Util.java [hunk 3]
```diff
@@ -1309,4 +1348,183 @@ public static JSONObject getNotesFromNoteFile(File notesFile) {
         }
         return jsonNotes;
     }
+
+
+    public static boolean isDownloading(Context context) {
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Util.java]: How about putting all the update stuff in it's own file and class to make it clear that this stuff is about updates? "isDownloading" in a Utils class is quite generic. The same goes for a number of the other methods added in this PR.

File: app/src/main/java/nz/org/cacophony/birdmonitor/Util.java [hunk 4]
```diff
@@ -1309,4 +1348,183 @@ public static JSONObject getNotesFromNoteFile(File notesFile) {
...
+        Matcher newVersionMatches = p.matcher(versionName);
+        int curV, newV;
+        if (newVersionMatches.matches() && currentMatches.matches()) {
+            for (int i = 0; i < currentMatches.groupCount(); i++) {
+                curV = Integer.parseInt(currentMatches.group(i+1));
+                if (i < newVersionMatches.groupCount()) {
+                    newV = Integer.parseInt(newVersionMatches.group(i+1));
+                    if (newV > curV) {
+                        return true;
+                    }else if(newV < curV){
```
Comment (mjs) [inline app/src/main/java/nz/org/cacophony/birdmonitor/Util.java]: spaces

File: app/src/main/java/nz/org/cacophony/birdmonitor/Util.java (hunk 2 above)
Comment (gferraro) [reply app/src/main/java/nz/org/cacophony/birdmonitor/Util.java:857]: It is retrieving seconds but we want milliseconds.
